execute/result.go

```diff
@@ -68,8 +69,10 @@ func (s *result) Do(f func(flux.Table) error) error {
 				return msg.err
 			}
 			if err := f(msg.table); err != nil {
+				msg.table.RefCount(-1)
```

Comment:
Why does only this error branch decrement the counter?

Reply:
There's another line under this one for decrementing it in the normal situation. Let me refactor this so it uses a defer and anonymous function.